spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/orm/jpa/HibernateJpaAutoConfigurationTests.java

```diff
@@ -600,7 +600,7 @@ EventCapturingApplicationListener jpaUsingApplicationListener(EntityManagerFacto
 
 		static class EventCapturingApplicationListener implements ApplicationListener<ApplicationEvent> {
 
-			private final List<ApplicationEvent> events = new ArrayList<>();
```

Comment:
I don't think this change is necessary. Within the context of the two tests that use the listener, events are only ever delivered on the main thread.